src/Type/IntersectionType.php

```diff
@@ -221,28 +221,38 @@ public function acceptsWithReason(Type $otherType, bool $strictTypes): AcceptsRe
 	}
 
 	public function isSuperTypeOf(Type $otherType): TrinaryLogic
+	{
+		return $this->isSuperTypeOfWithReason($otherType)->result;
+	}
+
+	public function isSuperTypeOfWithReason(Type $otherType): AcceptsResult
 	{
 		if ($otherType instanceof IntersectionType && $this->equals($otherType)) {
-			return TrinaryLogic::createYes();
+			return AcceptsResult::createYes();
 		}
 
 		if ($otherType instanceof NeverType) {
-			return TrinaryLogic::createYes();
+			return AcceptsResult::createYes();
 		}
 
-		return TrinaryLogic::createYes()->lazyAnd($this->getTypes(), static fn (Type $innerType) => $innerType->isSuperTypeOf($otherType));
+		return AcceptsResult::createYes()->and(...array_map(static fn (Type $innerType) => $innerType->isSuperTypeOfWithReason($otherType), $this->types));
 	}
 
 	public function isSubTypeOf(Type $otherType): TrinaryLogic
+	{
+		return $this->isSubTypeOfWithReason($otherType)->result;
+	}
+
+	public function isSubTypeOfWithReason(Type $otherType): AcceptsResult
 	{
 		if (($otherType instanceof self || $otherType instanceof UnionType) && !$otherType instanceof TemplateType) {
-			return $otherType->isSuperTypeOf($this);
+			return $otherType->isSuperTypeOfWithReason($this);
 		}
 
-		$result = TrinaryLogic::lazyMaxMin($this->getTypes(), static fn (Type $innerType) => $otherType->isSuperTypeOf($innerType));
+		$result = AcceptsResult::maxMin(...array_map(static fn (Type $innerType) => $otherType->isSuperTypeOfWithReason($innerType), $this->types));
```

Comment:
Here we're loosing a `lazyMaxMin ` into a `maxMin`.

I dunno if the lazy could be kept on AcceptsResults but I saw similar lazy lost when `isAcceptedWithReasonBy` was introduced (In this same class for instance).